pkg/apiserver/apiserver_test.go

```diff
@@ -785,6 +796,105 @@ func TestDelete(t *testing.T) {
 	}
 }
 
+func TestDeleteWithOptions(t *testing.T) {
+	storage := map[string]RESTStorage{}
+	simpleStorage := SimpleRESTStorage{}
+	ID := "id"
+	storage["simple"] = &simpleStorage
+	handler := handle(storage)
+	server := httptest.NewServer(handler)
+	defer server.Close()
+
+	grace := int64(300)
+	item := &api.DeleteOptions{
+		GracePeriodSeconds: &grace,
+	}
+	body, err := codec.Encode(item)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	client := http.Client{}
+	request, err := http.NewRequest("DELETE", server.URL+"/api/version/simple/"+ID, bytes.NewReader(body))
+	res, err := client.Do(request)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if res.StatusCode != http.StatusOK {
```

Comment:
Yes, but that requires a bigger change to the C/U/D methods.  Wanted to sync with @lavalamp because tried this once before and didn't like it.  Was considering changing all three mutator signatures to:

    func Method(ctx, obj) (runtime.Object, api.Status, err)

The second return value indicates status that modified the runtime object - status, return code (accepted vs created vs ok vs redirect).  Delete could return nil runtime.Object instead of requiring consumers to type check the returned object.  Client methods could also return this.  It doesn't have to be api.Status - it could be something similar.  Goal is to clearly delineate error case (go style) vs object (first ret val) vs restful modifiers (second ret val).  We could wrap the runtime object but that requires consumers to unwrap which seems like bad OO


> On Mar 18, 2015, at 9:36 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/apiserver/apiserver_test.go:
> 
> > +	grace := int64(300)
> > +	item := &api.DeleteOptions{
> > +		GracePeriodSeconds: &grace,
> > +	}
> > +	body, err := codec.Encode(item)
> > +	if err != nil {
> > +		t.Fatalf("unexpected error: %v", err)
> > +	}
> > +
> > +	client := http.Client{}
> > +	request, err := http.NewRequest("DELETE", server.URL+"/api/version/simple/"+ID, bytes.NewReader(body))
> > +	res, err := client.Do(request)
> > +	if err != nil {
> > +		t.Fatalf("unexpected error: %v", err)
> > +	}
> > +	if res.StatusCode != http.StatusOK {
> Is there any reason why we'd want to return different status codes (accepted) for graceful deletion vs. immediate deletion?
> 
> —
> Reply to this email directly or view it on GitHub.
>